quipucords/api/tests_networkprofile.py

```diff
@@ -104,6 +104,43 @@ def test_create_empty_host(self):
              'ssh_port': '22',
              'credentials': [self.cred_for_upload]})
 
+    def test_create_valid_hosts(self):
+        """Test valid host patterns."""
+
+        self.create_expect_201(
+            {'name': 'netprof1',
+             'hosts': ['10.10.181.9',
+                       '10.10.181.9/16',
+                       '10.10.128.[1:25]',
+                       '10.10.[1:20].25',
+                       '10.10.[1:20].[1:25]',
+                       'localhost',
+                       'mycentos.com',
+                       'my-rhel[a:d].company.com',
+                       'my-rhel[120:400].company.com'],
+             'ssh_port': '22',
+             'credentials': [self.cred_for_upload]})
+
+    def test_create_invalid_hosts(self):
+        """An empty string is not a host identifier."""
+
+        self.create_expect_400(
+            {'name': 'netprof1',
+             'hosts': ['192.1..2',
+                       '192.01.5.10',
+                       '192.01.5.1/',
+                       '192.01.5.[1:10]/10',
+                       '192.3.4.455',
+                       '192.3.4.455/16',
+                       '10.10.[181.9',
+                       '10.10.128.[a:25]',
+                       '10.10.[1-20].25',
+                       '1.1.1.1/33',
+                       'my_rhel[a:d].company.com',
+                       'my-rhel[a:400].company.com'],
+             'ssh_port': '22',
+             'credentials': [self.cred_for_upload]})
```

Comment:
Maybe we should have a (smaller) test that checks the error message is as expected?